src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -449,21 +448,28 @@ CodeWriter emitAndIndent(String s) throws IOException {
       if (trailingNewline) {
         emitIndentation();
         if (javadoc) {
-          out.append(" * ");
+          append(" * ");
         } else if (comment) {
-          out.append("// ");
+          append("// ");
         }
       }
-
-      out.append(line);
+      append(line);
       trailingNewline = false;
     }
     return this;
   }
 
-  private void emitIndentation() throws IOException {
+  private void emitIndentation() {
     for (int j = 0; j < indentLevel; j++) {
-      out.append(indent);
+      append(indent);
+    }
+  }
+
+  private void append(CharSequence csq) {
+    try {
+      out.append(csq);
+    } catch (IOException e) {
```

Comment:
True. But it is small and compact one against a scattered solution effecting all `*Name` and `*Spec` classes. And, imho, the first proposal just transformed that "small complexity" into an unchecked exception - it wasn't lost, as the `JavaFile.writeTo()` still reported `IOException`s.

As an internal in-memory `StringBuilder out` is a no-go anyway, this PR is more like a nice-to-have for the import refactoring mentioned in #409.